vyper/semantics/types/subscriptable.py

```diff
@@ -101,8 +101,12 @@ class _SequenceT(_SubscriptableT):
 
     def __init__(self, value_type: VyperType, length: int):
 
+        # length check added to existing child classes DArrayT and SArrayT
+        # in order to show the user source code for debugging.
+        # if you've hit this it is unexpected behavior and likely
+        # a new child class has been created which needs a similar check
```

Comment:
i think this comment is not really needed, or can be reduced to
```suggestion
    # sanity check
```
the rest is pretty clear from the fact that it's a panic.